From 170ff5e2e8d3e81475df856fc463e09760ab11c7 Mon Sep 17 00:00:00 2001
From: "Owen W. Taylor" <otaylor@fishsoup.net>
Date: Thu, 7 Aug 2014 10:21:51 +0200
Subject: [PATCH 3/3] MetaTextureTower: actually mark revalidated levels as
 valid

We were never actually clearing the invalid area on the smaller levels
of the texture tower, so we recomputed those levels every time.

https://bugzilla.gnome.org/show_bug.cgi?id=734400
---
 src/compositor/meta-texture-tower.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/compositor/meta-texture-tower.c b/src/compositor/meta-texture-tower.c
index 5d82add..80675c2 100644
--- a/src/compositor/meta-texture-tower.c
+++ b/src/compositor/meta-texture-tower.c
@@ -432,6 +432,9 @@ texture_tower_revalidate (MetaTextureTower *tower,
                           int               level)
 {
   texture_tower_revalidate_fbo (tower, level);
+
+  tower->invalid[level].x1 = tower->invalid[level].x2 = 0;
+  tower->invalid[level].y1 = tower->invalid[level].y2 = 0;
 }
 
 /**
-- 
2.1.0.rc1

